server/game/core/card/Card.ts

```diff
@@ -359,16 +357,20 @@ export class Card extends OngoingEffectSource {
     // ******************************************* KEYWORD HELPERS *******************************************
     /** Helper method for {@link Card.keywords} */
     protected getKeywords() {
-        const keywords = [...this.printedKeywords];
-
-        for (const gainedKeyword of this.getOngoingEffectValues(EffectName.GainKeyword)) {
-            keywords.push(gainedKeyword);
+        let keywords = [...this.printedKeywords];
```

Comment:
This works but I want to offer an alternative design suggestion for your consideration based on what I was thinking of doing for ability blanking. Keyword blanking is much simpler so there's definitely flexibility in the approaches and this does work, just want to run a suggestion past you.

One of the design principles I tried to use in this repo (with varying levels of success) is isolating out the complexity of specific mechanics into their own classes as much as possible so that entities using those mechanisms didn't have to worry about the mechanics. This code is a big mechanics-heavy, I think there's a way we could isolate it a bit so that the code here is really simple. It does have some tradeoffs though.

A way you could do this differently would be to add a list called `BlankingEffects` to the `KeywordInstance` class. Then, whenever a blanking effect is added to a unit, it adds itself to the BlankingEffects list of each keyword on that unit (even if already blanked). It could also have an `isBlank` property that's just a boolean returning whether BlankingEffects is empty or not.

Then here the check is just filtering for keywords that have `isBlank = false`. Obviously we're not reducing complexity that much overall since the effect has to add itself to the unit's keywords which implies some new methods and functionality, and it has to do bookkeeping to remove itself from those keywords when it's unapplied. But overall I think it would make the functionality here a little easier to read and consolidate the core logic into one place.

That said, some of the same benefits come if we just refactor this code into a few more sub-methods. I'm at peace with whichever you prefer.

Reply:
I gave your idea a shot. It works and it does indeed seem very clean.